extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/ConfigurationService.java

```diff
@@ -223,4 +223,36 @@ public int getMaxResults() throws GuacamoleException {
         );
     }
 
+    /**
+     * Returns whether or not LDAP aliases will be dereferenced,
+     * as configured with guacamole.properties.
+     * By default they will never be dereferenced.
+     *
+     * @return
+     *     An integer representing the status of of alias
+     *     dereferencing, as configured in guacamole.properties.
+     *
+     * @throws GuacamoleException
+     *     If guacamole.properties cannot be parsed.
+     */
+    public int getDereferenceAliases() throws GuacamoleException {
+        String derefAliases = environment.getProperty(
+            LDAPGuacamoleProperties.LDAP_DEREFERENCE_ALIASES,
+            "never"
```

Comment:
Okay, the latest commit takes a stab at an implementation of the new enum property.  I referenced the EncryptionMethod and EncryptionMethodProperty classes in doing the implementation.  It's slightly different since I'm mapping actual values for the JLDAP library instead of using encryption method to determine default port, but I think it'll do the trick.

Also, I looked up the OpenLDAP values for the DEREF setting in ldap.conf, and it matches JLDAP exactly, so I'm going to stick with the never/searching/finding/always values at this point in time, unless someone would like to suggest other options.  Looks like it's more standard than I have experienced - I usually just toggle between never and always.